### PR TITLE
Export script (Node version) - Setup (PR)

### PR DESCRIPTION
**Pull request by didoo**
## :pushpin: Summary

Initial setup of the following tools (this solves #161):

- **[TypeScript](https://www.typescriptlang.org/)**
- **[eslint](https://eslint.org/)**

An open question is if we want to add also [prettier](https://prettier.io/), but I'd prefer to discuss with the team if we want to adopt it too (for me, yes).

_(Notice: this is a cherripick of 508a5af0d09d8d743ea01d4ae095f9e3440f5c24 from #155)_

### How to review this PR.

Look at the diff in GitHub. 

### Things to look at:

- `package.json` - is the right format? do we want to add authors? other info?
- `tsconfig.json` - is everything ok here?
- `.eslintrc.js` - is everything ok here?

## :link: External Links

Issue: #161

## :building_construction: How to Build and Test the Change

If you want to test the setup, pull the branch and then:

```
// install dependencies
cd ~/[project-path]/script && yarn

// test typesafety
yarn typecheck

// code linting
yarn lint

// run a mock script
yarn sync
```

## :+1: Definition of Done

- [ ] package.json correctly set-up
- [ ] TypeScript added and correctly configured
- [ ] eslint added and correctly configured
- [ ] Prettier added (optional)

### :speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

Examples: 
- issue (ux,non-blocking): These buttons should be red, but let's handle this in a follow-up.
- suggestion (non-blocking): Let's change this wording to make it easier to understand.
- issue (blocking): We shouldn't introduce this kind of tech debt; let's pair and resolve the issue in a more sustainable way.

### CODE REVIEW

**Comment by didoo** [comment]
@MelSumner @amyrlam I have asked @Dhaulagiri to review this small PR (in the context of the larger Node export script project). If haven't added you as reviewers, but if you have time (and opinions!) on this, I would like to have your feedback (among other things, if we want to introduce also _prettier_ or not).

File: scripts/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "flight-icons-scripts",
+  "version": "0.0.1",
+  "private": true,
+  "main": "index.js",
+  "license": "MIT",
```
Comment (Dhaulagiri) [inline scripts/package.json:6]:
```suggestion
  "license": "MPL-2.0",
```

**Review by Dhaulagiri — APPROVED** [review]
this seems reasonable to me. I don't have a lot of experience with typescript to offer any valuable feedback, so I'd defer to your experience there.

As to the question of prettier I say yes but also that we could add it in subsequent PRs, particularly if we find a default config we like.

**Comment by amyrlam** [comment]
Prettier is great for me! 

FWIW, we have Prettier in ember-flight-icons out of the box, but not in CI yet it looks like from these links:
https://github.com/hashicorp/flight/blob/main/ember-flight-icons/package.json#L47
https://github.com/hashicorp/flight/blob/main/.github/workflows/ci-ember-flight-icons.yml#L81
https://github.com/hashicorp/flight/blob/main/ember-flight-icons/package.json#L92-L100

We may want one Prettier to absorb the other, I vaguely remember doing this elsewhere.

Don't care re: Prettier defaults, a pre-commit hook can help with that if ppl disagree. Not suggesting it now just throwing in my two cents

**Pull request merged**